Add documentation on how to use Weblate for translations

erral opened the pull request:
Now we can use [Weblate](https://hosted.weblate.org/projects/plone/) to translate Plone, so I have added a section in the How to contribute page explaining it.

<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1687.org.readthedocs.build/

<!-- readthedocs-preview plone6 end -->

erral committed:
document how to contribute using Weblate

erral committed:
add volto note

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
```diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
 
+(contributing-plone-core-translations-using-weblate)=
+
+## Using Weblate to translate Plone
```
```suggestion
## Weblate for translations
```

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
```diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
 
 4. Commit your changes, and create a pull request.
 
+(contributing-plone-core-translations-using-weblate)=
```
```suggestion
(contributing-weblate-for-translations)=
```

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
```diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
+
+## Using Weblate to translate Plone
+
+[Weblate](https://weblate.org/) is an Open Source project to help software developers translate their projects.
```
```suggestion
[Weblate](https://weblate.org/) is an open source project to help software developers translate their projects.
```

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
```diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
+
+[Weblate](https://weblate.org/) is an Open Source project to help software developers translate their projects.
+
+It lets the translator to work in a web based interface, without the need of installing any third-party software or without the need to use git or GitHub.
```
```suggestion
Translators can work in a web interface, and not have to install third-party software or use git or GitHub.
```

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
```diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
+
+It lets the translator to work in a web based interface, without the need of installing any third-party software or without the need to use git or GitHub.
+
+Plone Foundation has obtained a so called `libre` account for Plone, meaning we get the hosting with many other Open Source Projects at the [Hosted](https://hosted.weblate.org/) platform that Weblate offers to Open Source projects.
```
```suggestion
The Plone Foundation has obtained a "gratis Libre plan" account for Plone.
Plone gets free hosting at the [Hosted](https://hosted.weblate.org/) platform that Weblate offers to open source projects.
```

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
```diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
+
+Plone Foundation has obtained a so called `libre` account for Plone, meaning we get the hosting with many other Open Source Projects at the [Hosted](https://hosted.weblate.org/) platform that Weblate offers to Open Source projects.
+
+This means that the translator can go to the [Plone project and Weblate](https://hosted.weblate.org/projects/plone/) and start translating there.
```
```suggestion
Translators can go to the [Plone project in Weblate](https://hosted.weblate.org/projects/plone/) to write translations.
```

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
```diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
+
+This means that the translator can go to the [Plone project and Weblate](https://hosted.weblate.org/projects/plone/) and start translating there.
+
+You will need to create an account there and then your translations will be automatically contributed back to Plone's GitHub repository.
```
```suggestion
Translators will need to create an account there.
Then your translations will be automatically contributed back to Plone's GitHub repository.
```

stevepiercy commented on docs/i18n-l10n/contributing-translations.md:
````diff
@@ -77,6 +77,24 @@ The process of translating the Volto frontend is the following.
+You will need to create an account there and then your translations will be automatically contributed back to Plone's GitHub repository.
+
+```{note}
+At this moment Volto translations can't be contributed using Weblate. We are working on this issue and we hope to have it fixed as soon as possible.
````
```suggestion
At this moment Volto translations can't be contributed using Weblate.
The Volto Team are working on this issue, and hope to fix it as soon as possible.
```

stevepiercy reviewed (dismissed):
This is great news! Thanks for your persistence to see it through for core. Looking forward to Volto.

I made a few changes to make statements more concise. I also looked up the full name of the plan on Weblate.

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

erral committed:
Update docs/i18n-l10n/contributing-translations.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

mauritsvanrees reviewed (dismissed):
LGTM.

FTR, I tried registering on weblate with my GitHub account, but nothing happens, the site just keeps waiting for a response.

erral committed:
Merge branch '6.0' into erral-6.0-weblate-for-translations

erral commented:
> LGTM.
> 
> FTR, I tried registering on weblate with my GitHub account, but nothing happens, the site just keeps waiting for a response.

I have checked the configuration on the Weblate side and there is nothing special enabled to allow/deny users login in, it is only required to log in in weblate.

Moreover, if you check some PRs that have been merged using the Weblate integration, GitHub already links the Weblate account with the GitHub account.

stevepiercy commented:
I took Weblate for a spin, and resolved a few failing checks for Spanish in the `plone` component. What happens next?

> Then your translations will be automatically contributed back to Plone's GitHub repository.

That kind of scares me. What if I translate "plone" to "💩"?

In any case, I did not see a PR get opened or a commit pushed to any Plone GitHub repo. What did I do wrong, or are the docs incorrect?

mauritsvanrees commented:
Maybe add something like: "Authenticating via GitHub does not work.  It is best to create an account on Weblate itself."  I have done so now, and then it works fine.

I have updated four Dutch translations (the ones marked as needing an edit, which seems to translate to translations marked as "fuzzy" in the po file).  I am left with the same question as Steve, on how this now continues and if I need to do something myself.

Ah, I see a branch [`translations-plone`](https://github.com/collective/plone.app.locales/compare/master...translations-plone) on GitHub.  I guess Weblate automatically pushes changes there.  Would be good to know how often this happens.  Maybe a few times per day?

@stevepiercy I see your changes have made it to that branch in [this commit](https://github.com/collective/plone.app.locales/commit/08ce32eec2c4461b857b237451ba4418476dc126).

@erral Do you then manually merge this branch every week or so, or only before making a release and before updating the po files to match the latest source code of the Plone packages?  Some documentation on that would be good to have, so someone other than you can take over when you win the lottery. :-)
Not sure if this part of the documentation is the best part for that, although it may answer a few questions that new translators have.  There could be a section for that, and then the current section could end with: "If you want to know what happens after you have finished your translations, read the next section."

erral commented:
Ok I will extend the docs.

Yes the changes are pushed by weblate from time to time, and I Check amd merge the branch weekly more or less

stevepiercy committed:
Update contributing-translations.md

Using our combined super powers, add information from @mauritsvanrees and @erral to clarify the process.

stevepiercy commented:
@erral @mauritsvanrees I grabbed your comments and wrote up the documentation a bit more, breaking the introduction and the how-to guide into separate sections. Please read through it, and let me know. Thank you!

This is kickass! Thanks for working through this.

erral reviewed (commented):
Simple. Perfectly explained.

Thanks!

stevepiercy commented:
Pull request preview: https://plone6--1687.org.readthedocs.build/i18n-l10n/contributing-translations.html#weblate-for-translations.

stevepiercy committed:
Fix redirect to weblate.org

stevepiercy commented:
We can ignore the failing linkchecks, as I created PRs to fix those:

- https://github.com/plone/volto/pull/6239
- https://github.com/plone/documentation/pull/1689

stevepiercy commented:
Thanks @erral and @mauritsvanrees for making this happen. Always a pleasure to work with y'all.
